storage/cache/subtree_cache.go

```diff
@@ -128,38 +92,28 @@ func NewSubtreeCache(strataDepths []int, populateSubtree storage.PopulateSubtree
 	}
 
 	return &SubtreeCache{
-		stratumInfo:         sInfo,
+		layout:              layout,
 		populate:            populateSubtree,
 		populateConcurrency: *populateConcurrency,
 		prepare:             prepareSubtreeWrite,
 	}
 }
 
-func (s *SubtreeCache) stratumInfoForNodeID(id storage.NodeID) stratumInfo {
-	return s.stratumInfoForPrefixLength(id.PrefixLenBits - 1)
-}
-
-func (s *SubtreeCache) stratumInfoForPrefixLength(l int) stratumInfo {
-	return s.stratumInfo[l/depthQuantum]
-}
-
 // preload calculates the set of subtrees required to know the hashes of the
 // passed in node IDs, uses getSubtrees to retrieve them, and finally populates
 // the cache structures with the data.
 func (s *SubtreeCache) preload(ids []storage.NodeID, getSubtrees GetSubtreesFunc) error {
 	// Figure out the set of subtrees we need.
 	want := make(map[string]storage.NodeID)
 	for _, id := range ids {
-		sInfo := s.stratumInfoForNodeID(id)
-		pxKey := id.PrefixAsKey(sInfo.prefixBytes)
+		pxID := s.layout.GetSubtreeRoot(id)
```

Comment:
If we got the key directly here then it would remove another access to the internals of a NodeID.

Reply:
I started doing this and realised that there is still a couple of places below where we need `NodeID` (for example, the `getSubtrees` function signature). So if we return a prefix string here, we sill convert it back to `NodeID` assuming it's aligned to bytes, and hence there is still a dependency. I will look into breaking this dependency separately.